### PR TITLE
[17.0][MIG] report_substitute: Migration to 17.0

### PR DESCRIPTION
**Pull request by lef-adhoc**

### CODE REVIEW

**Comment by lef-adhoc** [comment]
Hi @sbejaoui, may you have check this ?
Thank you

**Review by sbejaoui — APPROVED** [review]
LGTM, code review

**Review by atchuthan — CHANGES_REQUESTED** [review]
Please add changes available in https://github.com/adhoc-dev/oca-reporting-engine/pull/1

Code migration is not complete when we used the initial commit, we made corrections and sent PR.

File: report_substitute/models/mail_thread.py [hunk 1]
```diff
@@ -0,0 +1,20 @@
+# Copyright 2019 ACSONE SA/NV
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import models
+
+
+class MailThread(models.AbstractModel):
+    _inherit = "mail.thread"
+
+    def message_post_with_template(self, template_id, **kwargs):
```
Comment (matiasperalta1) [inline report_substitute/models/mail_thread.py]: Este método cambió 
ahora es message_post_with_source
ver en este commit (https://github.com/odoo/odoo/commit/4775bd93a2bf9947236f4f7f06d6f8cb7734f7ef)

**Review by atchuthan — APPROVED** [review]
:+1:

**Comment by sakthi-shan** [comment]
@sbejaoui can we get this merged?

**Comment by sakthi-shan** [comment]
@lef-adhoc  
DeprecationWarning: XML declarations in HTML module descriptions are deprecated since Odoo 17, report_substitute can just have a UTF8 description with not need for a declaration.
We are getting this Warning in the log with the latest code. Please make the changes.

File: report_substitute/models/mail_thread.py [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+        source_ref,
+        render_values=None,
+        message_type="notification",
+        subtype_xmlid=False,
+        subtype_id=False,
+        **kwargs,
+    ):
+        template, view = self._get_source_from_ref(source_ref)
+        old_report_template_ids = False
+        if template.report_template_ids and self.ids:
```
Comment (SodexisTeam) [inline report_substitute/models/mail_thread.py]: We are getting this error, when template is false.
```
  File "/opt/odoo/sodtest17-odoo/src/community_modules_links/report_substitute/models/mail_thread.py", line 21, in message_post_with_source
    if template.report_template_ids and self.ids:
AttributeError: 'bool' object has no attribute 'report_template_ids'
```

**Review by SodexisTeam — CHANGES_REQUESTED** [review]

**Comment by lef-adhoc** [comment]
@SodexisTeam Could you check if it is ok now?

**Review by SodexisTeam — APPROVED** [review]
Tested and it looks good :+1:

**Comment by atchuthan** [comment]
@OCA/reporting-engine-maintainers can we merge this PR?

**Pull request merged**